common/browser_context_test.go

```diff
@@ -50,3 +51,240 @@ func TestNewBrowserContext(t *testing.T) {
 		assert.True(t, k6ObjScriptFound, "k6ObjScript was not initialized in the context")
 	})
 }
+
+func TestFilterCookies(t *testing.T) {
```

Comment:
Do we need a `t.Parallel` in the main body of the test and also in the sub tests?